libwild/src/args.rs

```diff
@@ -716,11 +722,33 @@ impl Args {
         parse(input)
     }
 
-    pub(crate) fn setup_thread_pool(&self) -> Result {
+    /// Sets up the thread pool, using the explicit number of threads if specified,
+    /// or falling back to the jobserver protocol if available.
+    /// and return the number of token which will be returned at drop place.
+    ///
+    /// https://www.gnu.org/software/make/manual/html_node/POSIX-Jobserver.html
+    pub(crate) fn setup_thread_pool(&mut self) -> Result<Vec<Acquired>> {
+        let mut tokens = Vec::new();
+        self.available_threads = self.num_threads.unwrap_or_else(|| {
+            // SAFETY: should be called early before other descriptors are opened
```

Comment:
How sure do you think we can be that we haven't opened any files at this point? I guess one other option would be to create the client earlier, perhaps even before argument parsing, then pass it in. I'm happy to make this change if you don't want to.

Reply:
I tried checked what environment variables cargo is passing to rustc. For me at least, it looks like `CARGO_MAKEFLAGS=-j --jobserver-fds=8,9 --jobserver-auth=8,9`. So, unfortunately it seems to be using file descriptors, not named pipes.

Reply:
Thanks for the testing effort. I decided to create an abstraction where `Args` need to be promoted to `ActivatedArgs`, allowing `Client` to be constructed early but activated at a specific point.
What do you think about it?